llvm/lib/Transforms/Vectorize/VPlanPatternMatch.h

```diff
@@ -149,12 +149,22 @@ struct is_zero_int {
   bool isValue(const APInt &C) const { return C.isZero(); }
 };
 
+struct is_one {
+  bool isValue(const APInt &C) const { return C.isOne(); }
+};
+
 /// Match an integer 0 or a vector with all elements equal to 0.
 /// For vectors, this includes constants with undefined elements.
+/// This is different from PatternMatch::m_Zero(), which also matches null
```

Comment:
There's the corresponding PatternMatch::m_ZeroInt, so you can just strip the comment I think.